### ISSUE TITLE
Media not attaching if participant disconnects and reconnects

### ISSUE THREAD

**Issue by camroncade** [twilio/twilio-video.js#62]
If I have two users, let's say User 1 and User 2.

If U2 disconnects and then reconnects, on U1's browser, the remote media U2 does not attach. If you dial down into the the remote participants media object, it shows that it is attached (it shows the correct div, that it is attached) but there is no `<video>` or `<audio>` element actually in the DOM.

On U2's side, both the local media and the remote media attachments are working correctly.

I've tested and found this behavior in both Chrome and Firefox.

Additionally, if a new user U3 were to join the room, that remote media attaches just fine. It only seems to show this issue when the same user disconnects and then reconnects.

Any ideas?

**Comment by camroncade** [comment]
Update:

I've discovered if U2 refreshes the browser (thus generating a new access token in my application), and then rejoins the room, the media attaches correctly on U1.

**Comment by diydan** [comment]
+1  Same issue here intermittently

**Comment by markandrus** [comment]
Hi @camroncade,

Sorry it's taken a while to respond to your issue.

> If U2 disconnects and then reconnects, on U1's browser, the remote media U2 does not attach. If you dial down into the the remote participants media object, it shows that it is attached (it shows the correct div, that it is attached) but there is no <video> or <audio> element actually in the DOM.

This suggests to me that, although the Media object is attached, no Tracks have been added internally (probably you would also see that the Participant's Media's `.tracks` Map is empty and that no "trackAdded" events fire, either).

> I've discovered if U2 refreshes the browser (thus generating a new access token in my application), and then rejoins the room, the media attaches correctly on U1.

Hmm, this is strange, as the Access Token needs only to be valid at the time of connect. The Access Token also controls whether or not TURN can be used, although that shouldn't affect whether or not the Track objects get constructed.

Since you mention Media, I take it you are using one of the Beta versions of twilio-video.js. Is there any way you could test with the 1.0.0 release? We fixed a handful of Track-related bugs since.

@diydan would you please open a new issue? We can follow up in more detail there.

Thanks,
Mark

**Comment by Tanktiger** [comment]
I had a similiar issue with Version 1.0.0
I tested my Application once on Win10 with Chrome and Win8 with Chrome. I got Video and Sound, also after reconnect.
On the other day i want to test it with my friend on MacOS with Chrome and Win8 with Chrome. Video appears but audio does not working.
We tested Skype to check the microfones and everything worked.
So i tested with my first test enviroment and now there is the same problem.

As code i am using the codes from the guides without any changes
